Source/WPF/MyMoney/Database/Money.cs

```diff
@@ -8460,10 +8466,15 @@ public IList<Transaction> GetTransactionsBySecurity(Security s, Predicate<Transa
                     t.RoutingPath = "|";
                 }
 
-
+                // The UnitPrice is not a mandatory field.  Add rarely have it and dividend never have it.

```

Comment:
Herein lies a big problem.  The reason I never attempted to show a historical value of your investments accounts is because we do not have enough data, we do not have all the historical stock values.  So the data you are computing here is just plain wrong.  There's no point showing a misleading chart.  To show a historical trend of the value of an investment account is a useful feature to have, but I want to do it right, which means getting historical stock prices for the range of the graph we are showing then compute the real value of holding taking stock splits into account, using something like our CostBasisCalculator class tracking all buys, sells and stock splits.  In order to get all historical stock prices we should be able to use the same service I'm using already in StockQuotes.cs.  Of course we'd want to then cache these values someplace, need a new SQL table for that so we don't keep requesting the same stock history over and over.  So doing all this properly is big job, which is why I haven't tackled it yet...  Fidelity already does these charts plenty well, and so I didn't see huge value in trying to rebuild all that here.  Another way to implement this would be to build a "Server" process that gets the stock price every day, then we wouldn't need to go get a long history of prices, except once to build that back history.

Reply:
Today the UI shows a trend graph for stocks.  It's totally wrong as it does not show values related to the stock, but instead outsating balances in the accounts that have that stock.  That was very misleading.
One option was to take out the trend graph altogether for stocks, which I though about, but I was not ready to make a big change like that.
Making work with the data that exists actaully gave reasonable results for me.  For 401K accounts, I have an updated stock price every 2 weeks.  For stocks with reinvestments, at least 4 stock prices per year.  So it gives some reasonable graph over long periods of time.
It is very correct that getting prices for each day would be the right solution.
This change was meant to be a small change to get my feet wet that did not require data storage changes.

Reply:
Ok, I understand, lets see if we can wrap this up.  I'll try your changes and test them out on my side.

Reply:
I tested your latest code, looks good.